packages/wallet-management/src/walletAutomation.ts

```diff
@@ -1,67 +1,52 @@
 /* eslint-disable radix */
+import { Web3Provider } from '@ethersproject/providers';
 import type { StaticToken } from '@lifi/sdk';
 import {
   MetaMaskProviderErrorCode,
   getChainById,
   prefixChainId,
 } from '@lifi/sdk';
-import type { Provider } from './types';
 
 export const switchChain = async (
-  provider: Provider,
+  provider: Web3Provider,
   chainId: number,
 ): Promise<boolean> => {
-  return new Promise((resolve, reject) => {
-    try {
-      provider
-        .request({
-          method: 'wallet_switchEthereumChain',
-          params: [{ chainId: getChainById(chainId).metamask?.chainId }],
-        })
-        .catch((error: any) => {
-          if (error.code !== MetaMaskProviderErrorCode.userRejectedRequest) {
-            addChain(provider, chainId).then((result) => resolve(result));
-          } else {
-            reject(error);
-          }
-        });
-      provider.once('chainChanged', (id: string) => {
-        if (parseInt(id) === chainId) {
-          resolve(true);
-        }
-      });
-    } catch (error: any) {
-      // const ERROR_CODE_UNKNOWN_CHAIN = 4902
-      if (error.code !== MetaMaskProviderErrorCode.userRejectedRequest) {
-        addChain(provider, chainId).then((result) => resolve(result));
-      } else {
-        console.error(error);
-        resolve(false);
-      }
+  try {
+    await provider.send('wallet_switchEthereumChain', [
+      { chainId: getChainById(chainId).metamask?.chainId },
+    ]);
```

Comment:
Why we moved from `provider.request` to `provider.send`?

Reply:
The provider object changed its type. The new type does not have request. It has send. 
This change was necessary to facilitate wallet automations without knowing which wallet is currently being used. Not knowing which wallet is being used happens when an external walletmanagement has been set but no wallet automation functions.